### PR TITLE
MP-8366 - Database configuration for "User" 

### PR DESCRIPTION
**Pull request by danield-ap**
- Fix typo in TemperatureTestField
- Fix UserTypeOptions namespace to `Nox.Types`

### CODE REVIEW

File: src/Nox.EntityFramework.Postgres/PostgresUserDatabaseConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using Nox.Types;
+using Nox.Types.EntityFramework.Types;
+
+namespace Nox.EntityFramework.Postgres;
+
+public class PostgresUserDatabaseConfiguration : UserDatabaseConfigurator
```
Comment (dejobratic) [inline src/Nox.EntityFramework.Postgres/PostgresUserDatabaseConfiguration.cs:6]: Why do we need this configuration?

Postgres stores text data in text type, which is recommended anyways.

**Review by dejobratic — COMMENTED** [review]
Looks good. Just a question for the Postgres configuration.

File: src/Nox.EntityFramework.Postgres/PostgresUserDatabaseConfiguration.cs (hunk 1 above)
Comment (gjorgji-grgovski-authoritypartners) [reply src/Nox.EntityFramework.Postgres/PostgresUserDatabaseConfiguration.cs:6]: We do not need this database-specific configuration.

I tested this with 3 different attributes
image
which were created like this in the SqlServer database
image
and like this in the Postgres database
image

File: src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        bool isKey)
+    {
+        var userOptions = property.UserTypeOptions ?? new UserTypeOptions();
+
+        builder
+            .Property(property.Name)
+            .IsRequired(property.IsRequired)
+            .HasMaxLength(userOptions.MaxLength)
+            .If(userOptions.MaxLength == userOptions.MinLength, builder2 => builder2.IsFixedLength())
+            .IfNotNull(GetColumnType(userOptions), b => b.HasColumnType(GetColumnType(userOptions)))
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs:27]: This is not needed

File: src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            .HasConversion<UserConverter>();
+    }
+
+    public string GetKeyPropertyName(NoxSimpleTypeDefinition key) => key.Name;
+    public virtual string? GetColumnType(UserTypeOptions typeOptions)
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs:32]: This is not needed

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

File: src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs (hunk 2 above)
Comment (danield-ap) [reply src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs:27]: `IfNotNull` line removed.

File: src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs (hunk 3 above)
Comment (danield-ap) [reply src/Nox.Types.EntityFramework/Types/User/UserDatabaseConfigurator.cs:32]: virtual method removed.

**Review by rochar — APPROVED** [review]
Looks good 
Next we can add it to all Nox typed

**Review by dejobratic — APPROVED** [review]
looks good

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Pull request merged**